src/Controller/Traits/RoleManagementTrait.php

```diff
@@ -0,0 +1,124 @@
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright Copyright 2010 - 2018, Cake Development Corporation (https://www.cakedc.com)
+ * @license MIT License (http://www.opensource.org/licenses/mit-license.php)
+ */
+
+namespace CakeDC\Users\Controller\Traits;
+
+use Cake\Core\Configure;
+use Cake\Validation\Validator;
+use CakeDC\Users\Exception\UserNotActiveException;
+use CakeDC\Users\Exception\UserNotFoundException;
+use CakeDC\Users\Exception\WrongPasswordException;
+use CakeDC\Users\Plugin;
+use Exception;
+
+/**
+ * Covers the password management: reset, change
+ *
+ * @property \Cake\Http\ServerRequest $request
+ */
+trait RoleManagementTrait
+{
+
+    /**
+     * Change Role
+     * Can be used by superadmin to change user roles
+     *
+     * @param int|string|null $id user_id, null for logged in user id
+     * @return mixed
+     */
+    public function changeRole($id = null)
+    {
+
+        $identity = $this->getRequest()->getAttribute('identity');
+        $identity = $identity ?? [];
+        $userId = $identity['id'] ?? null;
+
+        if ($userId) {
+            if ($id && $this->CanUserEditRole($identity)) {
+                // superuser update user roles
+                $user = $this->getUsersTable()->get($id);
+                $configRoles = Configure::read('Users.AvailableRoles');
+                $availableRoles = [];
+                foreach ($configRoles as $role) {
+                    $availableRoles[$role] = $role;
+                }
+                $redirect = ['action' => 'index'];
+            } else {
+                $this->Flash->error(
+                    __d('cake_d_c/users', 'Changing role is not allowed')
+                );
+                return $this->redirect(Configure::read('Users.Profile.route'));
+
+                return;
+            }
+        } else {
+            $this->Flash->error(
+                __d('cake_d_c/users', 'Login to perform this action')
+            );
+            return $this->redirect(Configure::read('Users.Profile.route'));
+        }
+
+        if ($this->getRequest()->is(['post', 'put'])) {
+            try {
+                if (!in_array($this->getRequest()->getData()['role'], $availableRoles)) {
+                    throw new Exception('Invalid role supplied');
+                }
+                $user = $this->getUsersTable()->patchEntity(
+                    $user,
+                    $this->getRequest()->getData(),
+                    [
+                        'accessibleFields' => [
+                            'role' => true,
+                        ],
+                    ]
+                );
+
+
+                if ($user->getErrors()) {
+                    $this->Flash->error(__d('cake_d_c/users', 'Role could not be changed'));
+                } else {
+                    $user->is_superuser = $user->role === 'superuser';
```

Comment:
Not sure about setting is_superuser based on role string, maybe we should add other form input. what do you think @steinkel ?

Reply:
`is_superuser` and `role` are using 2 different approaches for authorization. I would not change one based on another.

Reply:
Both are handled in authorization plugin but with specific policy classes. SuperuserPolicy for  `is_superuser` and RbacPolicy for `role`

Reply:
If we would want them to be independent, I think a checkbox is a better option, let me know If i should add that in

Reply:
I agree about adding a checkbox @steinkel what do you think?

Reply:
ok to me too